Python_04/main.py

```diff
@@ -0,0 +1,183 @@
+            weight = input('Enter weight - ')
+    weight = int(weight)
+    sp.append({login: [sex, height, weight]})
+
+def log_sp(sp, log = []):
+    log = list.copy(sp)
+    print('Users:')
+    for i in range(len(log)):
+        for k, v in log[i].items():
+            print(' ', k)
+    return log
+
+def info(inf_req):
+    log = list.copy(sp)
+    for i in range(len(log)):
+        for k, v in log[i].items():
+            if k == inf_req:
+                os.system('cls') 
+                print(k, ':', '\nSex - ', v[0], '\nHeight - ', v[1], '\nWeight - ', v[2])
+                BMI = round((v[2]/(v[1]**2))*10000, 2)
+                print('BMI - ', BMI)
+                if BMI > 60 or BMI < 10:
+                    print('Impossible value')
+                else:
+                    BMI = round(BMI)
+                    skale = '10' + '='*48 + '60'
+                    print(skale[:BMI-10], '|', skale[BMI-10:], sep='=')
+                break
+            else:
+                if i + 1 == len(log):
+                    os.system('cls')
+                    print('Fail')
+        if k == inf_req:
+            break
+            
+    
+def change(n_req):
+    log = list.copy(sp)
+    for i in range(len(log)):
+        for k, v in log[i].items():  
+            if k == n_req:  
+                slot_chg = input('Change(sex, height, weight)? - ')
+                if slot_chg == 'sex':
+                    info_chg = input('Enter new sex(Male/Female) - ')
+                    while info_chg != 'Male' and info_chg != 'Female':
+                        print('Fail')
+                        info_chg = input('Enter new sex(Male/Female) - ')   
+                    else: 
+                        v[0] = info_chg
+                    print('Changed')
+                elif slot_chg == 'height':
+                    info_chg = input('Enter new height - ')
+                    while type(info_chg) != int:
+                        try:
+                            info_chg = int(info_chg)
+                        except:
+                            print('Fail')
+                            info_chg = input('Enter height - ')
+                    v[1] = int(info_chg)
+                    print('Changed')
+                elif slot_chg == 'weight':
+                    info_chg = input('Enter new weight - ')
+                    while type(info_chg) != int:
+                        try:
+                            info_chg = int(info_chg)
+                        except:
+                            print('Fail')
+                            info_chg = input('Enter weight - ')
+                    v[2] = int(info_chg)
+                    print('Changed')
+                else:
+                    print('Fail')
+            else:
+                if i + 1 == len(log):
+                    os.system('cls')
+                    print('Fail')
+        if k == n_req:
+            break
+
+def delete(sp, del_item):
+    log = list.copy(sp)
```

Comment:
it looks complicated...
the logic could be easier if users were stored in a dictionary
```python
{
 1: { "name": "Ivan"},
 2: { "name": "Petia"}
}
```